tablecodec/tablecodec.go

```diff
@@ -441,6 +441,7 @@ func unflatten(datum types.Datum, ft *types.FieldType, loc *time.Location) (type
 				return datum, errors.Trace(err)
 			}
 		}
+		datum.SetUint64(0)
```

Comment:
Do we need to clean the `i` for TypeDuration?

Reply:
No, we needn't. `d.SetMysqlDuration(x)` will set `i` again.

Reply:
@crazycs520 `SetDuration -> SetMysqlDuration`. Inside `SetMysqlDuration`, we assign the value of duration to `i` after type conversion. So we definitely need clean `i` of `Datum` in the case of `TypeDuration`.